crates/jstz_cli/src/config.rs

```diff
@@ -194,10 +201,63 @@ pub struct SandboxConfig {
     pub pid: u32,
 }
 
-pub const SANDBOX_OCTEZ_NODE_PORT: u16 = 18731;
-pub const SANDBOX_OCTEZ_NODE_RPC_PORT: u16 = 18730;
-pub const SANDBOX_JSTZ_NODE_PORT: u16 = 8933;
-pub const SANDBOX_OCTEZ_SMART_ROLLUP_PORT: u16 = 8932;
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum NetworkName {
+    Custom(String),
+    // Dev network uses sandbox config
+    Dev,
+}
+
+impl fmt::Display for NetworkName {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            NetworkName::Custom(name) => write!(f, "{}", name),
+            NetworkName::Dev => write!(f, "dev"),
+        }
+    }
+}
+
+impl FromStr for NetworkName {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s.to_lowercase().as_str() {
+            "dev" => Ok(NetworkName::Dev),
+            other => Ok(NetworkName::Custom(other.to_string())),
+        }
+    }
+}
+
+impl Serialize for NetworkName {
+    fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for NetworkName {
+    fn deserialize<D>(deserializer: D) -> core::result::Result<NetworkName, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s: String = Deserialize::deserialize(deserializer)?;
+        NetworkName::from_str(&s).map_err(serde::de::Error::custom)
+    }
+}
```

Comment:
I'm fairly certain you can configure serde derive to serialize/deserialize based on `to/from_str`.